src/SAWScript/Crucible/MIR/Builtins.hs

```diff
@@ -252,6 +254,61 @@ mir_postcond term = MIRSetupM $ do
   loc <- SS.toW4Loc "mir_postcond" <$> lift (lift getPosition)
   Setup.crucible_postcond loc term
 
+mir_points_to ::
+  SetupValue {- ^ LHS reference -} ->
+  SetupValue {- ^ RHS value -} ->
+  MIRSetupM ()
+mir_points_to ref val =
+  MIRSetupM $
+  do cc <- getMIRCrucibleContext
+     loc <- getW4Position "mir_points_to"
+     st <- lift get
+     let env = MS.csAllocations (st ^. Setup.csMethodSpec)
+         nameEnv = MS.csTypeNames (st ^. Setup.csMethodSpec)
+
+     allocIdx <-
+       case ref of
+         MS.SetupVar idx -> pure idx
+         _ -> X.throwM $ MIRPointsToNonReference ref
+
+     referentTy <- mir_points_to_check_lhs_validity ref loc
+     valTy <- typeOfSetupValue cc env nameEnv val
+     unless (checkCompatibleTys referentTy valTy) $
+       fail $ unlines
+         [ "Referent type incompatible with value in `mir_points_to` statement:"
+         , "  Referent type: " ++ show (PP.pretty referentTy)
+         , "  Value type:    " ++ show (PP.pretty valTy)
+         ]
+
+     tags <- view Setup.croTags
+     let md = MS.ConditionMetadata
+              { MS.conditionLoc = loc
+              , MS.conditionTags = tags
+              , MS.conditionType = "MIR points-to"
+              , MS.conditionContext = ""
+              }
+     Setup.addPointsTo (MirPointsTo md allocIdx [val])
+
+-- | Perform a set of validity checks on the LHS reference value in a
+-- 'mir_points_to' command. In particular:
+--
+-- * Check that the LHS is in fact a valid reference type.
+--
+-- Returns the 'Mir.Ty' that the LHS points to.
+mir_points_to_check_lhs_validity ::
+  SetupValue {- ^ LHS reference -} ->
+  W4.ProgramLoc {- ^ The location in the program -} ->
+  Setup.CrucibleSetupT MIR TopLevel Mir.Ty
+mir_points_to_check_lhs_validity ref loc =
+  do cc <- getMIRCrucibleContext
+     st <- get
+     let env = MS.csAllocations (st ^. Setup.csMethodSpec)
+         nameEnv = MS.csTypeNames (st ^. Setup.csMethodSpec)
+     refTy <- typeOfSetupValue cc env nameEnv ref
+     case refTy of
+       Mir.TyRef referentTy _ -> pure referentTy
+       _ -> throwCrucibleSetup loc $ "lhs not a reference type: " ++ show refTy
```

Comment:
This code checks if the reference passed to `mir_points_to` is actually a reference type.